### PR TITLE
update docs and dependencies

### PR DESCRIPTION
**Pull request by Dav-C**
Remove the secrets-providers app as a dependency
Update readme to fix icons
Add/Update docs dependencies to match core
Increase Nautobot version for local development and CI to 2.3.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,44 +21,44 @@ Regardless, the Onboarding App greatly simplifies the onboarding process by allo
...
-| Platform                | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
-| Manufacturer            | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
-| Serial Number           | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
-| Device Type             | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
-| Mgmt Interface          | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
-| Mgmt IP Address         | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |
+| Hostname                | &#9989; | &#9989; | &#9989; | &#9989; | &#9989; | &#9989; | &#x274C; |
```
Comment (glennmatthews) [inline README.md:24]: Can we use the actual unicode/emoji characters  ✅ ❌ 🧪 here? For example:

```suggestion
| Hostname                | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ❌ |
```

(see also #218).

File: changes/223.housekeeping [hunk 2]
```diff
@@ -0,0 +1 @@
+Remove secrets-provider app dependency update doc dependencies
```
Comment (glennmatthews) [inline changes/223.housekeeping:1]:
```suggestion
Removed secrets-provider app dependency.
Updated documentation dependencies.
```

File: nautobot_device_onboarding/tests/test_basic.py [hunk 3]
```diff
@@ -11,11 +11,21 @@ class TestDocsPackaging(unittest.TestCase):
...
+            "dependencies"
+        ]
+        with open(
+            f"{parent_path}/docs/requirements.txt", "r", encoding="utf-8"
+        ) as file:
+            requirements = [
+                line
+                for line in file.read().splitlines()
+                if (len(line) > 0 and not line.startswith("#"))
+            ]
```
Comment (glennmatthews) [inline nautobot_device_onboarding/tests/test_basic.py:28]: Is this just a formatting change? Did the poetry.lock updates include updates to ruff, or why was this necessary?

File: tasks.py [hunk 4]
```diff
@@ -66,11 +66,15 @@ def is_truthy(arg):
 
 
 def _is_compose_included(context, name):
```
Comment (glennmatthews) [inline tasks.py:68]: Are all of the below changes just due to a different version of ruff?

Comment (Dav-C) [reply tasks.py:68]: The scope of this PR is too large and includes things that overlap with drift manager. I will close it.
